### PR TITLE
Add missing documentation for expressions and environments.

### PR DESCRIPTION
**Pull request by robertoraggi**
Add missing documentation for Expr and Env. Also, start to use reference
declaration using `@link` and hide some internal symbols.

### CODE REVIEW

File: @here/harp-datasource-protocol/lib/Expr.ts [hunk 1]
```diff
@@ -195,31 +197,34 @@ interface ReferenceResolverState {
...
+     * The scope of an {@link Expr} used as dynamic property attribute value.
      */
     Dynamic
 }
 
+// tslint:disable:max-line-length
 /**
- * Abstract class defining a shape of a [[Theme]]'s expression
+ * Abstract class representing the
+ * {@link https://github.com/heremaps/harp.gl/blob/master/%40here/harp-datasource-protocol/StyleExpressions.md | style expressions}
```
Comment (nzjony) [inline @here/harp-datasource-protocol/lib/Expr.ts:222]: can this link to a local file? You are anyway in the `harp-datasource-protocol` lib

File: @here/harp-datasource-protocol/lib/Expr.ts [hunk 2]
```diff
@@ -195,31 +197,34 @@ interface ReferenceResolverState {
...
     Condition,
 
     /**
-     * The scope of an [[Expr]] used as dynamic property attribute value.
+     * The scope of an {@link Expr} used as dynamic property attribute value.
      */
     Dynamic
 }
 
+// tslint:disable:max-line-length
```
Comment (nzjony) [inline @here/harp-datasource-protocol/lib/Expr.ts:219]: this disables it for the rest of the file, is that what you want?

File: @here/harp-datasource-protocol/lib/Expr.ts [hunk 3]
```diff
@@ -626,7 +642,7 @@ export class MatchExpr extends Expr {
 }
 
 /**
- * @hidden
+ * A node representing a `case` expression.
```
Comment (nzjony) [inline @here/harp-datasource-protocol/lib/Expr.ts:645]: would it make sense to link to the syntax of this expression?

Comment (robertoraggi) [reply @here/harp-datasource-protocol/lib/Expr.ts:645]: all the expressions follow the same syntax and the link to the style expression is in the dcoumentation of the base Expr class.

File: @here/harp-datasource-protocol/lib/Expr.ts (hunk 2 above)
Comment (robertoraggi) [reply @here/harp-datasource-protocol/lib/Expr.ts:219]: oops, how can i disable the max-line-length only for the comment?

**Pull request merged**